robot-agent/config.js

```diff
@@ -1,18 +1,32 @@
 const fs = require('fs');
 const dotenv = require('dotenv');
+const { clone } = require('@transitive-sdk/utils');
 
 dotenv.config({path: './.env'});
 dotenv.config({path: './.env_user'});
 
 global.config = {};
 const fleetConfig = {};
 
-try {
-  global.config = JSON.parse(fs.readFileSync('./config.json', {encoding: 'utf8'}));
-  console.log(`Using config:\n${JSON.stringify(global.config, true, 2)}`);
-} catch (e) {
-  console.log('No config.json file found or not valid JSON, proceeding without.');
-}
+const refreshGlobalConfigFromFile = () => {
+  try {
+    global.config = JSON.parse(fs.readFileSync('./config.json', {encoding: 'utf8'}));
+    console.log(`Using config:\n${JSON.stringify(global.config, true, 2)}`);
+    if (global.data) {
+      // update the config in the fleet data store
+      console.log(`Updating global config in ${global.AGENT_PREFIX}/info/config`);
+      global.data.update(`${global.AGENT_PREFIX}/info/config`, clone(global.config));
+    }
+  } catch (e) {
+    console.log('No config.json file found or not valid JSON, proceeding without.');
+  }
+};
+refreshGlobalConfigFromFile();
+
+fs.watch('./config.json', (curr, prev) => {
```

Comment:
`(curr, prev)` seems to still be left over from the previous use of watchFile, but it's very confusing, because that's not what the `watch` callback receives: https://nodejs.org/docs/latest-v20.x/api/fs.html#fswatchfilename-options-listener